deployment_scripts/appimage/build_appimage.sh

```diff
@@ -0,0 +1,225 @@
+  cp -r $REPO_MONKEY_SRC/common $INSTALL_DIR
+  cp -r $REPO_MONKEY_SRC/monkey_island $INSTALL_DIR
+  cp ./run_appimage.sh $INSTALL_DIR/monkey_island/linux/
+  cp ./island_logger_config.json $INSTALL_DIR/
+  cp ./server_config.json.standard $INSTALL_DIR/monkey_island/cc/
+
+  # TODO: This is a workaround that may be able to be removed after PR #848 is
+  # merged. See monkey_island/cc/environment_singleton.py for more information.
+  cp ./server_config.json.standard $INSTALL_DIR/monkey_island/cc/server_config.json
+}
+
+install_monkey_island_python_dependencies() {
+  log_message "Installing island requirements"
+
+  requirements_island="$ISLAND_PATH/requirements.txt"
+  # TODO: This is an ugly hack. PyInstaller is a build-time dependency and should
+  #		  not be installed as a runtime requirement.
+  sed '4d' $requirements_island | sponge $requirements_island
+
+  ${python_cmd} -m pip install -r "${requirements_island}"  --ignore-installed --prefix /usr --root=$APPDIR || handle_error
+}
+
+download_monkey_agent_binaries() {
+log_message "Downloading monkey agent binaries to ${ISLAND_BINARIES_PATH}"
+  mkdir -p "${ISLAND_BINARIES_PATH}" || handle_error
+  curl -L -o ${ISLAND_BINARIES_PATH}/${LINUX_32_BINARY_NAME} ${LINUX_32_BINARY_URL}
+  curl -L -o ${ISLAND_BINARIES_PATH}/${LINUX_64_BINARY_NAME} ${LINUX_64_BINARY_URL}
+  curl -L -o ${ISLAND_BINARIES_PATH}/${WINDOWS_32_BINARY_NAME} ${WINDOWS_32_BINARY_URL}
+  curl -L -o ${ISLAND_BINARIES_PATH}/${WINDOWS_64_BINARY_NAME} ${WINDOWS_64_BINARY_URL}
+
+  # Allow them to be executed
+  chmod a+x "$ISLAND_BINARIES_PATH/$LINUX_32_BINARY_NAME"
+  chmod a+x "$ISLAND_BINARIES_PATH/$LINUX_64_BINARY_NAME"
+}
+
+install_mongodb() {
+  log_message "Installing MongoDB"
+
+  mkdir -p $MONGO_PATH
+  "${ISLAND_PATH}"/linux/install_mongo.sh ${MONGO_PATH} || handle_error
+}
+
+generate_ssl_cert() {
+  log_message "Generating certificate"
+
+  chmod u+x "${ISLAND_PATH}"/linux/create_certificate.sh
+  "${ISLAND_PATH}"/linux/create_certificate.sh ${ISLAND_PATH}/cc
+}
+
+build_frontend() {
+    pushd "$ISLAND_PATH/cc/ui" || handle_error
+    npm install sass-loader node-sass webpack --save-dev
+    npm update
+
+    log_message "Generating front end"
+    npm run dist
+    popd || handle_error
+}
+
+build_appimage() {
+	log_message "Building AppImage"
+	appimage-builder --recipe monkey_island_builder.yml --log DEBUG --skip-appimage
+
+	# There is a bug or unwanted behavior in appimage-builder that causes issues
+	# if 32-bit binaries are present in the appimage. To work around this, we:
+	#   1. Build the AppDir with appimage-builder and skip building the appimage
+	#   2. Add the 32-bit binaries to the AppDir
+	#   3. Build the AppImage with appimage-builder from the already-built AppDir
+	#
+	# Note that appimage-builder replaces the interpreter on the monkey agent binaries
+	# when building the AppDir. This is unwanted as the monkey agents may execute in 
+	# environments where the AppImage isn't loaded.
```

Comment:
This means that monkeys can only be executed on the island or what?

Reply:
Ahh, this is the explanation to why you download the binaries after building the AppDir

Reply:
Yep. There are better solutions to this problem, but since we're not using AppImage Builder in the long run, they're not worth investing in.